packages/messaging/ios/RNFBMessaging/RNFBMessaging+FIRMessagingDelegate.m

```diff
@@ -49,6 +49,8 @@ - (void)observe {
 
 // JS -> `onTokenRefresh`
 - (void)messaging:(FIRMessaging *)messaging didReceiveRegistrationToken:(NSString *)fcmToken {
+  if(fcmToken == nil) // Don't crash when the token is reset
+    return;
```

Comment:
Just a whitespace nit - adding a JS API might not be so hard, I think the reason it doesn't exist is because it doesn't have any corresponding APIs on the javascript API and the public API surface area for this package is usually defined by the JS API. Not sure what the policy is on non-JS interfaces but as long as it was coherent across iOS and Android to match the rest of the Instance ID API (none of which has corresponding JS interfaces) it would probably go in

```suggestion
  if (fcmToken == nil) { // Don't crash when the token is reset
    return;
  }
```